crowbar_framework/app/models/service_object.rb

```diff
@@ -1193,6 +1205,17 @@ def apply_role(role, inst, in_queue)
           tmprole = RoleObject.find_role_by_name "#{role_name}_remove"
           unless tmprole.nil?
             elem_remove = tmprole.name
+            # save remove intention in #{@bc_name}-databag
+            databag["deployment"][@bc_name]["elements"][elem_remove] ||= []
+
+            old_nodes.each do |old_node|
+              @logger.debug "saving #{elem_remove} intention for #{old_node}"
+              unless databag["deployment"][@bc_name]["elements"][elem_remove].include? old_node
+                databag["deployment"][@bc_name]["elements"][elem_remove] << old_node
+              end
+            end
+
+            databag.save
```

Comment:
> will saving like this trigger a bump in the revision, which would result in making the proposal not appear as applied for the latest revision?

@vuntz
do you mean the `crowbar-revision` ?
as i observed it counts up quite a few times on each apply or deactivate (12 times in my case with deactivating ceilometer). I don't know the exact logic behind it yet, all i know is that in deactivated state, it is yellow and says "has been changed since last deployment". applying it again it is green and up to date.